ratis-server/src/main/java/org/apache/ratis/server/storage/FileChunkReader.java

```diff
@@ -47,11 +51,14 @@ public class FileChunkReader implements Closeable {
    *
    * @param info the information of the file.
    * @param relativePath the relative path of the file.
+   * @param chunkMaxSize maximum chunk size for each {@link #readFileChunk()} call
    * @throws IOException if it failed to open the file.
    */
-  public FileChunkReader(FileInfo info, Path relativePath) throws IOException {
+  public FileChunkReader(FileInfo info, Path relativePath, int chunkMaxSize) throws IOException {
     this.info = info;
     this.relativePath = relativePath;
+    this.chunkMaxSize = chunkMaxSize;
```

Comment:
Would it make sense to limit this if file is smaller, i.e. `min(chunkMaxSize, info.getFileSize())`?

Reply:
You are right! We can save some unnecessary memory allocations.